contractcourt/chain_watcher_test.go

```diff
@@ -512,11 +512,29 @@ func TestChainWatcherLocalForceCloseDetect(t *testing.T) {
 		// since she's the one creating the HTLCs (lower balance). In
 		// order to simulate the commitment only having the remote
 		// party's output, we'll remove Alice's output.
-		if remoteOutputOnly {
+
+		// We now modify Alice's commitment tx outputs which will be
+		// used as the spending tx.
+		switch {
+		// Exclude Alice's local output.
+		case remoteOutputOnly && numUpdates > 0:
 			aliceCommit.TxOut = aliceCommit.TxOut[1:]
-		}
-		if localOutputOnly {
+
+		// Exclude Alice's remote output.
+		case localOutputOnly && numUpdates > 0:
 			aliceCommit.TxOut = aliceCommit.TxOut[:1]
+
+		// Exclude Alice's local output. When there're no updates,
+		// based on BIP69, Alice's to local output will be placed at
+		// index 1 since the pkscript is longer(32 bytes).
+		case remoteOutputOnly && numUpdates == 0:
```

Comment:
Is there an introduced failing test that caused these new cases to be added?

Reply:
Previously the outputs from the commit tx were not sorted, now it's sorted below, which is needed for testing the new revocation log.

Reply:
actually this is not needed, removed the commit.